Init db module

youagree opened the pull request.

Commit:
add db module

Commit:
init db structure

docent42 committed:
init scripts for all schema tables added. Create:tables,sequences,foreign keys,indexes

Commit:
remove unused files

youagree commented on physical-access-control-system-db/pom.xml:
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
добавить скрипт инициализации(schema squd) user squd password squd

docent42 replied:
Done

youagree commented on physical-access-control-system-db/pom.xml:
```diff
@@ -0,0 +1,59 @@
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>3.6.1</version>
```
подставить проперти

docent42 replied:
Done

youagree commented on physical-access-control-system-db/src/main/resources/database_changelog.xml:
```diff
@@ -0,0 +1,10 @@
+
+    <include file="changelog-v.1-cumulative.xml" relativeToChangelogFile="true"/>
+
+    <!-- todo relativeToChangelogFile, onFail(подобрать оптимальный параметр, прекондишены на каждое действие,-->
```
убрать todo

docent42 replied:
убрал

youagree commented on physical-access-control-system-db/src/main/resources/db/changelog_1/1-init_schema.xml:
```diff
@@ -0,0 +1,134 @@
+      constraintName="pk_employees_cars"
+      tableName="employees_cars"/>
+  </changeSet>
+
```
дополнительно сделать changeSet addUniqueKey

docent42 replied:
сделал

youagree commented on physical-access-control-system-db/src/main/resources/db/changelog_1/1-init_schema.xml:
```diff
@@ -0,0 +1,134 @@
+      <column name="creation_time" type="TIMESTAMP"/>
+      <column name="order_weight" type="DOUBLE"/>
+      <column name="real_weight" type="DOUBLE"/>
+      <column name="status" type="ENUM('CREATED','IN_PROGRESS','DONE','CANCELED')"/>
```
предлагаю сделать просто varchar

docent42 replied:
сделал варчар

youagree commented on physical-access-control-system-db/src/main/resources/db/changelog_1/1-init_schema.xml:
```diff
@@ -0,0 +1,134 @@
+    <comment>employees_cars table</comment>
+  </changeSet>
+
+  <changeSet author="ESV" id="employees_cars-fk-1">
```
перенести к месту, где создается эта таблица

docent42 replied:
здесь положение определено правилом: сначала таблицы , потом ключи

youagree commented on physical-access-control-system-db/src/main/resources/db/changelog_1/1-init_schema.xml:
```diff
@@ -0,0 +1,134 @@
+      <column name="status" type="ENUM('CREATED','IN_PROGRESS','DONE','CANCELED')"/>
+    </createTable>
+  </changeSet>
+
```
проверить на всякий случай, нужны ли нам индексы в этой таблице

youagree replied:
вопрос закрыт

youagree commented on physical-access-control-system-impl/src/main/resources/application.yml:
```diff
@@ -0,0 +1,18 @@
+spring:
+  application:
+    name: control-system
+  jpa:
```
добавить недостающие настройки для бд и попробовать запустить проект, чтобы хибернейт провалидировался

docent42 replied:
добавлено запущено провалидировано

youagree commented on physical-access-control-system-db/pom.xml:
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
попробовать накатить ликвибейс через плагин и после запустить сам проект

docent42 committed:
Code Review Notes 23-12-2020:
1.SQL sqript for db,user init
2.Liquibase version property added to dependency
3.TO_DO deleted from database_changelog.xml
4.Unique constraint added to cars_employees
5.VARCHAR type added to Order.status instead/
6.Datasource added to aplication.yaml
